gamla/transducer.py

```diff
@@ -0,0 +1,47 @@
+import functools
+from typing import Any, Callable, Iterable, TypeVar
+
+from gamla import currying, functional, functional_generic
+
+_S = TypeVar("_S")
+_X = TypeVar("_X")
+Reducer = Callable[[_S, _X], _S]
+Transducer = Callable[[Reducer], Reducer]
+
+
+def _transform_nth(n: int, f: Reducer):
+    @currying.curry
+    def transform_nth(step, state, x):
+        new_state_specific = f(state[n], x)
+        new_state = (*state[:n], new_state_specific, *state[n + 1 :])
+        final_state = step(new_state, x)
+        return final_state
+
+    return transform_nth
+
+
+def transjuxt(*funcs: Iterable[Transducer]) -> Transducer:
+    return functional_generic.pipe(
+        funcs,
+        enumerate,
+        functional_generic.map(functional.star(_transform_nth)),
+        functional.star(functional_generic.compose),
+    )
+
+
+@currying.curry
+def mapping(f: Callable[[Any], Any], step: Reducer):
```

Comment:
imo a bit confusing to have `gamla.map` and `gamla.mapping`. 

Maybe call them `transmap`, `transfilter`, `transcat`,etc. ?

Reply:
this is all pretty new, and eventually i think this is a better way to do `map` `filter` etc', so meanwhile I will just call it `map` and put it under the transducer namespace.